src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -114,29 +205,88 @@ private Index getIndex(String indexName) {
     return index;
   }
 
-  private Query buildQuery(String stringToMatch) {
+  private Query buildQuery(
+    String stringToMatch,
+    String difficulty,
+    Integer time
+  ) {
     Cursor responseCursor = Cursor.newBuilder().build();
     QueryOptions options = QueryOptions
       .newBuilder()
-      .setLimit(60)
+      .setLimit(10)
+      .setFieldsToSnippet("ingredients")
       .setSortOptions(
         SortOptions.newBuilder().setMatchScorer(MatchScorer.newBuilder())
       )
       .setCursor(responseCursor)
       .build();
 
-    Query query = Query
-      .newBuilder()
-      .setOptions(options)
-      .build(stringToMatch.toLowerCase());
+    String searchString = createSearchString(stringToMatch, difficulty, time);
+    Query query = Query.newBuilder().setOptions(options).build(searchString);
 
     return query;
   }
 
+  /**
+   * This function creates and returns the string used inside the search query.
+   */
+  private String createSearchString(
```

Comment:
Besides the comments I've wrote, which are for you to learn how to simplify conditions, you don't need to do things like:

!StringUtils.isBlank(stringToMatch) or !StringUtils.isBlank(difficulty)

Instead, you could always check for

!StringUtils.isBlank(searchString)

Basically, you need to prepend and AND if there's some text in the query already.